test_cases/scripts/intelligent_test_runner.py

```
@@ -272,8 +273,25 @@ def is_valid(self):
+            self.tasker_path = shutil.which("tasker")
+            if self.tasker_path is None:
+                # Fallback to tasker.py in current directory if not in PATH
+                if os.path.exists("./tasker.py"):
+                    self.tasker_path = "./tasker.py"
+                elif os.path.exists("../tasker.py"):
+                    # Also check parent directory (common when running from test_cases/)
+                    self.tasker_path = "../tasker.py"
+                else:
+                    # Last resort - use ./tasker.py and let it fail with clear error
+                    self.tasker_path = "./tasker.py"
+                    print(f"WARNING: Could not find 'tasker' in PATH or './tasker.py'")
```

Ruff (0.13.3)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)